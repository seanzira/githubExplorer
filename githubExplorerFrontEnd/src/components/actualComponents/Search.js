import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import '../componentStyles/Search.css';

// Search component
const Search = () => {

    // state to hold the username input from the user
    const [username, setUsername] = useState('');
    // useNavigate hook for programmatic navigation
    const navigate = useNavigate();

    // Function to handle the search from submission
    const handleSearch = async (e) => {
        e.preventDefault();
        navigate(`/user/${username}`);
    };

    return (
        <div className="search-container">
            <h1>Github Explorer</h1>
            <form onSubmit={handleSearch}>
                <input 
                    type='text'
                    placeholder='Search for a Github user'
                    value={username}
                    onChange={(e) => setUsername(e.target.value)}       
                />
                <button type='submit'>Search</button>
            </form>
        </div>
    );
};

// export the Search component for use in other parts of the application
export default Search;