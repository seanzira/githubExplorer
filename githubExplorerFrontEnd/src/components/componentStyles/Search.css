/* Main container for the search component */
div {
    background-color: black; 
    color: green;           
    padding: 20px;         
    border-radius: 8px;   
    text-align: center;     
    max-width: 400px;      
    margin: auto;          
}

/* Inner container to hold search elements */
.search-container {
    text-align: center;      
    padding: 20px;          
    border-radius: 8px;     
    max-width: 400px;       
}

/* Heading style */
h1 {
    font-size: 2rem;        
    margin-bottom: 20px;    
}

/* Input field style */
input {
    width: 100%;        
    padding: 10px;         
    border: 2px solid green; 
    border-radius: 4px;    
    background-color: black; 
    color: green;            
    font-size: 1rem;       
}

/* Input field focus style */
input:focus {
    outline: none;         
    border-color: lightgreen; 
}

/* Button style */
button {
    padding: 10px 20px;     
    background-color: green; 
    color: black;            
    border: none;            
    border-radius: 4px;     
    font-size: 1rem;      
    cursor: pointer;
    transition: background-color 0.3s;
}

/* Button hover style */
button:hover {
    background-color: lightgreen; 
}
