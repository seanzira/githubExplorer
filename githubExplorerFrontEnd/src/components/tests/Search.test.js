import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Search from '../actualComponents/Search';

describe('Search component', () => {
    it('matches snapshot', () => {
      const { asFragment } = render(
        <MemoryRouter>
          <Search />
        </MemoryRouter>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });