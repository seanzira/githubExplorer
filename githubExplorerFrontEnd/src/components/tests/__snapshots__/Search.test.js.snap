// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-container"
  >
    <h1>
      Github Explorer
    </h1>
    <form>
      <input
        placeholder="Search for a Github user"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
